lib/httpDigest.js

```diff
@@ -19,13 +22,54 @@ async function create({data, algorithm = 'sha256', useMultihash = true}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
-  const encodedData = new TextEncoder().encode(data);
+  algorithm = algorithm.toLowerCase();
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return _createMultiHash({digest});
+  }
+  return `SHA-256=${base64.encode(digest)}`;
+}
+
+/**
+ * Verifies digest of data created by create().
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.header - The digest header string to
+ *   be verified against.
+ *
+ * @returns {object} - Returns an object with verified true or false.
+ */
+async function verify({data, header}) {
+  if(typeof data !== 'string') {
+    data = JSON.stringify(data);
+  }
   let digest;
+  let encoded;
+  const algorithm = header.split('=')[0].replace('-', '').toLowerCase();
+  if(algorithm === 'sha256') {
+    encoded = header.substring(header.indexOf('=') + 1);
+    digest = await _getDigest({data, algorithm});
+  } else if(algorithm === 'mh') {
+    encoded = header.substring(header.indexOf('=') + 1).slice(1);
```

Comment:
How confident are we in these parsing rules? Is there a tiny http parsing lib we can rely on instead?

Reply:
Also, this isn't properly checking the multibase char for `u` as an indication of base64url encoding.

Reply:
> Also, this isn't properly checking the multibase char for u as an indication of base64url encoding.

+1, should test for that.

Reply:
> How confident are we in these parsing rules? Is there a tiny http parsing lib we can rely on instead?

Not super. (Obv we can get more confident with more tests.) But yeah, if there was a generic http header parsing lib (for that format specifically), might be better. (Not sure if there is one)

Reply:
If we're not fixing this now, let's make sure we've got good tests and add an issue for looking into a better/more confident parser lib/rules.